OpenRA.Game/Traits/Health.cs

```diff
@@ -122,7 +122,7 @@ public void InflictDamage(Actor self, Actor attacker, int damage, DamageWarhead
 				damage = Util.ApplyPercentageModifiers(damage, modifiers);
 			}
 
-			hp = Exts.Clamp(hp - damage, 0, MaxHP);
+			hp = hp - damage.Clamp(0, MaxHP);
```

Comment:
I think this has to be `hp = (hp - damage).Clamp(0, MaxHP);` to not change the calculation.